### PR TITLE
Dockerfile: install picolibc & ARM/RISC-V GCC via package manager

### PR DESCRIPTION
**Pull request by benpicco**
There seems to be a problem with the picolibc install in riotdocker. (https://github.com/RIOT-OS/riotdocker/pull/109#issuecomment-701505405)
Instead of trying to sort that out by ourself, just use the package [provided by Debian](https://packages.debian.org/search?suite=bullseye&searchon=names&keywords=picolibc). It has ~~no dependencies~~ the respective gcc packages as dependencies. 

So instead of manually downloading the toolchain, also install that from the repository.

### CODE REVIEW

**Comment by kaspar030** [comment]
didn't upstream miss risc-v support?

**Comment by benpicco** [comment]
I've been compile-testing for the `hifive1` with it the whole time.

https://packages.debian.org/testing/picolibc-riscv64-unknown-elf

**Comment by kaspar030** [comment]
somehow this failed *all* builds on travis.

**Comment by benpicco** [comment]
Yes because I was wrong about the packages having no dependencies.
They depend on the respective gcc package (e.g. https://packages.debian.org/bullseye/gcc-riscv64-unknown-elf) which we also compile from source.

File: Dockerfile [hunk 1]
```diff
@@ -162,6 +161,18 @@ RUN mkdir -p /opt && \
 
 ENV PATH $PATH:/opt/gnu-mcu-eclipse/riscv-none-gcc/${RISCV_VERSION}-${RISCV_BUILD}/bin
 
+# Install picolibc Debian package
+ARG PICOLIBC_VERSION=1.4.6-1
```
Comment (kaspar030) [inline Dockerfile:165]: there's a 1.7.3 package, would that work, too?

Comment (benpicco) [reply Dockerfile:165]: of course!

**Comment by kaspar030** [comment]
@benpicco should we push this? how can I help?

**Comment by benpicco** [comment]
Ah the problem is still that we install the compilers from outside the package manager, so the dependency of the picolibc package is not met.

For ARM, the 2019q4 version of GCC 9 we install manually is also available [on focal](https://packages.ubuntu.com/focal/gcc-arm-none-eabi), but for RISC-V our GCC 10.1.0-1.1 is a bit newer than what [focal has in store](https://packages.ubuntu.com/focal/gcc-riscv64-unknown-elf).

**Comment by benpicco** [comment]
Looks like it worked this time!

**Comment by benpicco** [comment]
bors try

**Comment by kaspar030** [comment]
ok, but we're switching to the ubuntu gcc packages completely. do we want that?

**Comment by benpicco** [comment]
bors try

**Comment by kaspar030** [comment]
I don't really like that this switches to ubuntu's version of arm gcc :/

**Comment by benpicco** [comment]
What's wrong with it?

**Comment by kaspar030** [comment]
> What's wrong with it?

how has it been compiled? we've been using the arm toolchain for years. and the ubuntu one used to have quirks (I remember the small enum thing? or was that debian?).

**Comment by benpicco** [comment]
I don't know, I've always just used the toolchain provided in the Ubuntu repository.

**Comment by benpicco** [comment]
I would expect that most users are installing the toolchain from their distribution's repository as it's customary on Linux.

I didn't have any issues with using it for the past years.

Mind you that the source we are pulling from is [deprecated](https://developer.arm.com/downloads/-/gnu-rm) and if you want to upgrade to a [new version](https://developer.arm.com/downloads/-/arm-gnu-toolchain-downloads) we'll have to adapt the build scripts.

I'm not keen on doing this tedious work when we could as well just install it from the repo instead.

**Comment by maribu** [comment]
If there are indeed issues with the Ubuntu toolchain, it would be IMO in our best interest to address them by getting fixes upstream into the Ubuntu repos. Ubuntu is very much the most popular Linux distro and Ubuntu user will use the toolchain that is packaged for Ubuntu. We have to deal with this in any case.

For me, it would also be easier to be able to reproduce the issues Ubuntu users have by just passing `BUILD_IN_DOCKER=1` instead of having to set up a container / VM with Ubuntu. And ideally I would like to run into the issues myself during the CI run, rather than having someone else run into it.

**Review by maribu — APPROVED** [review]

**Comment by kaspar030** [comment]
@maribu your call when to merge this (to not change riotdocker during release)

**Comment by kaspar030** [comment]
bors merge

**Comment by maribu** [comment]
bors merge

**Comment by kaspar030** [comment]
bors ping

**Comment by benpicco** [comment]
Thank you!

**Pull request merged**